core/routemgmt/createApi/createApi.js

```diff
@@ -0,0 +1,297 @@
+  var tenantInstance = message.tenantInstance || 'openwhisk';
+
+  // Log parameter values
+  console.log('GW URL        : '+message.gwUrl);
+  console.log('GW Auth       : '+utils.confidentialPrint(message.gwPwd));
+  console.log('namespace     : '+doc.namespace);
+  console.log('tenantInstance: '+message.tenantInstance);
+  console.log('API name      : '+doc.apiName);
+  console.log('basepath      : '+doc.gatewayBasePath);
+  console.log('relpath       : '+doc.gatewayPath);
+  console.log('GW method     : '+doc.gatewayMethod);
+  if (doc.action) {
+    console.log('action name: '+doc.action.name);
+    console.log('action namespace: '+doc.action.namespace);
+    console.log('action backendUrl: '+doc.action.backendUrl);
+    console.log('action backendMethod: '+doc.action.backendMethod);
+    console.log('action authkey: '+utils.confidentialPrint(doc.action.authkey));
+  }
+  console.log('apidoc        :\n'+JSON.stringify(doc , null, 2));
+
+  // Create and activate a new API path
+  // 1. Create tenant id this namespace.  If id exists, create is a noop
+  // 2. Obtain any existing configuration for the target API.  If none, this is a new API
+  // 3. Create the API document to send to the API GW.  If API exists, update it
+  // 4. Configure API GW with the new/updated API
+  var tenantId;
+  var gwApiId;
+  return utils.createTenant(gwInfo, doc.namespace, tenantInstance)
+  .then(function(tenant) {
+    console.log('Got the API GW tenant: '+JSON.stringify(tenant));
+    tenantId = tenant.id;
+    return Promise.resolve(utils.getApis(gwInfo, tenant.id, basepath));
+  })
+  .then(function(apis) {
+    console.log('Got '+apis.length+' APIs');
+    if (apis.length === 0) {
+      console.log('No APIs found for namespace '+doc.namespace+' with basepath '+basepath);
+      return Promise.resolve(utils.generateBaseSwaggerApi(basepath, doc.apiName));
+    } else if (apis.length > 1) {
+      console.error('Multiple APIs found for namespace '+doc.namespace+' with basepath '+basepath);
+      return Promise.reject('Internal error. Multiple APIs found for namespace '+doc.namespace+' with basepath '+basepath);
+    }
+    gwApiId = apis[0].id;
+    return Promise.resolve(utils.generateSwaggerApiFromGwApi(apis[0]));
+  })
+  .then(function(swaggerApi) {
+    if (doc.swagger) {
+      console.log('Use provided swagger as the entire API; override any existing API');
+      return Promise.resolve(doc.swagger);
+    } else {
+      console.log('Add the provided API endpoint');
+      return Promise.resolve(utils.addEndpointToSwaggerApi(swaggerApi, doc));
+    }
+  })
+  .then(function(swaggerApi) {
+    console.log('Final swagger API config: '+ JSON.stringify(swaggerApi));
+    return utils.addApiToGateway(gwInfo, tenantId, swaggerApi, gwApiId);
+  })
+  .then(function(gwApi) {
+    console.log('API GW configured with API');
+    var cliApi = utils.generateCliApiFromGwApi(gwApi).value;
+    console.log('createApi success');
+    return Promise.resolve(cliApi);
```

Comment:
Need to check here what get's return to the CLI when creating apis using swagger vs. creating one endpoint. And be sure we have a test cases in place.